packages/cli/src/shutdown/Shutdown.service.ts

```diff
@@ -0,0 +1,85 @@
+import { Container, Service } from 'typedi';
+import { ApplicationError, ErrorReporterProxy, assert } from 'n8n-workflow';
+import { Logger } from '@/Logger';
+
+export interface ServiceClass {
+	new (): Record<string, () => Promise<void> | void>;
+}
+
+export interface ShutdownHandler {
+	serviceClass: ServiceClass;
+	methodName: string;
+}
+
+/** Error reported when a listener fails to shutdown gracefully */
+export class ComponentShutdownError extends ApplicationError {
+	constructor(componentName: string, cause: Error) {
+		super('Failed to shutdown gracefully', {
+			level: 'error',
```

Comment:
I think that's the default?

Reply:
Could be. I'd prefer to keep it to make it explicit